### PR TITLE
refactor: metrics server. Simplify app.nim module

### PR DESCRIPTION
**Pull request by Ivansete-status**
## Description
Simplify `app.nim` and move the "metrics factory" procs to an already existing `waku_metrics` module

### CODE REVIEW

File: waku/node/waku_metrics.nim [hunk 1]
```diff
@@ -54,3 +55,35 @@ proc startMetricsLog*() =
...
+): Result[MetricsHttpServerRef, string] =
+  var metricsServer: MetricsHttpServerRef
+  if conf.metricsServer:
+    metricsServer = startMetricsServer(
+      conf.metricsServerAddress, Port(conf.metricsServerPort + conf.portsShift)
+    ).valueOr:
+      return
+        err("Starting metrics server failed. Continuing in current state:" & $error)
+
+  if conf.metricsLogging:
```
Comment (NagyZoltanPeter) [inline waku/node/waku_metrics.nim:86]: Just out of curiosity, is there a mean having metrics server running without metrics logging?

**Review by NagyZoltanPeter — APPROVED** [review]
LGTM!

File: waku/node/waku_metrics.nim (hunk 1 above)
Comment (gabrielmer) [reply waku/node/waku_metrics.nim:86]: I guess you can scrape the metrics for monitoring purposes without it polluting the logs?

**Review by gabrielmer — APPROVED** [review]
Amazingg, thank you!

**Pull request merged**